src/main/java/com/ibm/crypto/plus/provider/XDHPrivateKeyImpl.java

```diff
@@ -313,25 +313,60 @@ private byte[] processEncodedPrivateKey(byte[] encoded) throws IOException {
         byte[] keyBytes = inputValue[2].getOctetString();
         DerInputStream derStream = new DerInputStream(keyBytes);
         try {
-            // XDH private key in SunEC new Java 17 design requires [octet-string[octer-string[key-bytes]]] format,
-            // otherwise, it causes interop issue. JCK issue 569
-            this.key = derStream.getOctetString(); // Try J17 format [octet-string[octer-string[key-bytes]]]
-        } catch (IOException e) {
-            this.key = keyBytes; // Try J11 format [octer-string[key-bytes]]
+            // XDH private key in SunEC new Java 17 design requires [octet-string[octet-string[key-bytes]]] format,
+            // otherwise, it causes interop issue.
+            if (isCorrectlyFormedOctetString(keyBytes)) {
+                this.key = derStream.getOctetString(); // We know we are working with the format [octet-string[octet-string[key-bytes]]]
+            } else {
+                throw new IOException("Octet is malformed, use entire key byte array.");
```

Comment:
Do we need to throw the exception here? We just catch it and assign the key. I think this could just be an `else` statement.

Reply:
Yes i think this is better too as we dont use exceptions for flow control which is usually a bad practice.